python/tests/server/test_code_xfroms.py

```diff
@@ -0,0 +1,77 @@
+import os
+import pytest
+import sys
+import uuid
+import cog.code_xforms as code_xforms
+
+g_module_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason="requires python3.9 or higher")
```

Comment:
What requires Python 3.9 or higher? The test or the code? If it's the code then we definitely need a fallback mechanism because we have lots of models in production still running 3.8, and some running 3.7.

Reply:
code.